src/Fields/set!.jl

```diff
@@ -12,9 +12,7 @@ function set!(Φ::NamedTuple; kwargs...)
 end
 
 set!(u::AbstractField, v::Number) = parent(u) .= v
-
-set!(u::AbstractField{X, Y, Z, A}, v::AbstractField{X, Y, Z, A}) where {X, Y, Z, A} =
-    parent(u) .= parent(v)
+set!(u::AbstractField, v::AbstractField) = parent(u) .= parent(v)
```

Comment:
Ah, I think the type "diagonalization" was intentional --- we only want to resolve to `parent(u) .= parent(v)` when the fields are at the same location and on the same architecture.